examples/nodeprot-cli.rs

```diff
@@ -0,0 +1,175 @@
+use std::net::SocketAddr;
+
+use argh::FromArgs;
+use melnet2::{wire::tcp::TcpBackhaul, Backhaul};
+use themelio_nodeprot::{NodeRpcClient, Substate, TrustedHeight, ValClient};
+use themelio_structs::{Address, BlockHeight, NetID, Transaction, TxHash};
+use tmelcrypt::HashVal;
+
+fn main() {
+    smolscale::block_on(async move {
+        let args: Args = argh::from_env();
+        let backhaul = TcpBackhaul::new();
+        let rpc_client = NodeRpcClient(
+            backhaul
+                .connect(args.addr.to_string().into())
+                .await
+                .expect("failed to create RPC client"),
+        );
+
+        // one-off set up to test using a custom network.
+        let client = ValClient::new(args.netid, rpc_client);
+        let snapshot = client.insecure_latest_snapshot().await.unwrap();
```

Comment:
I had to add this method back, since it was the only way I could find lets me trust a custom network.